source/server/listener_manager_impl.cc

```diff
@@ -1014,6 +1008,15 @@ Network::DrainableFilterChainSharedPtr ListenerFilterChainFactoryBuilder::buildF
 
   auto& config_factory = Config::Utility::getAndCheckFactory<
       Server::Configuration::DownstreamTransportSocketConfigFactory>(transport_socket);
+  // The only connection oriented UDP transport protocol right now is QUIC.
+  const bool is_quic = listener_.udpListenerFactory() != nullptr &&
+                       !listener_.udpListenerFactory()->isTransportConnectionless();
+  if (is_quic && typeid(Quic::QuicServerTransportSocketConfigFactory) != typeid(config_factory)) {
```

Comment:
This is better. As a slight improvement, you could use the type URL of the factory proto type to avoid RTTI (style guide discourages reliance on RTTI).

Reply:
type_url has API version information. In this case it will be "type.googleapis.com/envoy.extensions.transport_sockets.quic.v3.QuicDownstreamTransport". If we deprecate v3 api, the test will break. 

Instead of typeid(), can I use dynamic_cast() == nullptr?

Reply:
@htuch WDYT?

Reply:
I wouldn't worry about deprecating v3 any time soon. Anyway, I don't feel that strongly. `dynamic_cast` is fine.

Reply:
I changed to use dynamic_cast instead. PTAL